src/components/CanvasAnimator.tsx

```diff
@@ -1,14 +1,36 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Frame, GraphIndex, LabeledGraph } from '../logic/commonInterfaces';
 import Canvas from './Canvas';
 
+const MS_PER_FRAME = 1000
+
 function CanvasAnimator({ graph, frames }
   : {
     graph: LabeledGraph,
     frames: Array<Frame>,
   }) {
+  const [animation, setAnimation] = useState({
+    started: true,
+    frameNum: 0,
+  });
+
+  useEffect(() => {
+    if (animation.started) {
+      setTimeout(() => {
+        let nextFrame = animation.frameNum + 1;
+        if (nextFrame === frames.length) {
+          nextFrame = 0;
+        }
+        setAnimation({ ...animation, frameNum: nextFrame });
```

Comment:
follow up to the previous comment: I think using
`setFrameNumber(...)`
would be cleaner

Reply:
Adding another method just to operate single variable is a bit much in my opinion.

Reply:
But this will not be a method, rather a hook